### PR TITLE
React Pagination

### LINKED ISSUE #1079: [BUILD] Pagination Component - React
**User Story:**
As a Spark User I want to be able to use React Components so that I have multiple options.

**AC:**
- Build Pagination Component - React
- Write Unit Test
- Complete Full Regression Test

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Adds the SprkPagination react component and fixes some typos related to SprkPromo.

### Associated Issue 
Closes #1079 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.
### Documentation
 - [x] Update Spark Docs React
 - [x] Update Component Sass Var/Class Modifier table

### Code
 - [x] Build Component in Spark React
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome

### CODE REVIEW

**Comment by yodasw16** [comment]
closes #1079

File: src/patterns/components/pagination/collection.yaml [hunk 1]
```diff
@@ -4,8 +4,10 @@ description: |
   It serves as navigation and informs users where they currently are and where they
   can go in the future.
 information:
-  - Spark-Core Pagination JS will take care of adding in the correct aria-label
-    to each link on first page load and when the items are tapped or clicked.
+  - As a general guideline, you should use the "default" variant when the pager has 3 or fewer pages,
+    and you should use the "long" variant when the pager has more than 3 pages.
+  - The React component will automatically switch between the "default" and "long" variants
```
Comment (afebbraro) [inline src/patterns/components/pagination/collection.yaml]: We'll want any React information under the React section, not in the main area. That away, if someone doesn't have "React" selected in the top page level code switcher, they don't see things related to React.

File: src/react/src/routes/SprkPromoDocs/SprkPromoDocs.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ const SprkPromoDocs = () => {
             additionalClasses="sprk-o-Stack--split@s"
             cta="link"
             ctaText="Learn More"
-            ctaLinkHref="https://www.sparkdesignsystem.com"
+            ctaHref="https://www.sparkdesignsystem.com"
```
[the remark this answers is not in the document]
Comment (bobwise) [reply src/react/src/routes/SprkPromoDocs/SprkPromoDocs.js]: Yes, it is `ctaHref` everywhere else.

File: src/react/projects/spark-core-react/src/SprkPagination/SprkPagination.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  idString: PropTypes.string,
+};
+
+SprkPagination.defaultProps = {
+  variant: 'default',
+  currentPage: 1,
+  additionalClasses: '',
+  nextLinkText: 'Next Page',
+  prevLinkText: 'Previous Page',
+  analyticsStringNext: '',
```
[the remark this answers is not in the document]
Comment (bobwise) [reply src/react/projects/spark-core-react/src/SprkPagination/SprkPagination.js]: I can see that being useful for sure. I'll add a single property and apply it to each of the individual pages.

I can't think of a good use case for allowing a custom property for every page though. "page_changed" would be useful, but idk about "page_2_clicked" ... "page_17_clicked" etc.

File: src/react/projects/spark-core-react/src/SprkHighlightBoard/SprkHighlightBoard.test.js [hunk 4]
```diff
@@ -35,6 +35,13 @@ it('should render an image if an imgSrc is provided', () => {
   expect(img.hasClass('sprk-c-HighlightBoard__image')).toBe(true);
 });
 
+it ('should render a header when provided', () => {
+  const wrapper = shallow(<SprkHighlightBoard heading='foo' />);
+  const header = wrapper.find('h1.sprk-c-HighlightBoard__heading');
+
+  expect(header.length).toBe(1);
```
Comment (afebbraro) [inline src/react/projects/spark-core-react/src/SprkHighlightBoard/SprkHighlightBoard.test.js]: Thanks for fixing coverage on HighlightBoard also!

File: src/react/src/components/SiteHeader/SiteHeader.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ function SiteHeader(props) {
   return (
     <header>
       <div className="drizzle-o-Layout__header sprk-u-mbn sprk-u-pvl drizzle-c-AngularHeader">
-        <Link to="/">
+        <Link to="/" className='sprk-b-Link'>
```
Comment (afebbraro) [inline src/react/src/components/SiteHeader/SiteHeader.js]: haha yes! Thanks for fixing this!

**Pull request merged**